docs/book/command-params.md

```diff
@@ -0,0 +1,203 @@
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class HelloCommand extends Command
+{
+    use InputParamTrait;
+
+    public static $defaultName = 'example:hello';
+
+    protected function configure() : void
+    {
+        $this->setName(self::$defaultName);
+        $this->addParam('name', 'Your name', InputParam::TYPE_STRING);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output) : int
+    {
+        $name = $this->getParam('name');
+        $output->writeln('Hello ' . $name . '!');
+
+        return 0;
+    }
+}
+```
+
+As you see it is very similar to adding option. As noted before there is no shortcut,
+so we always must use the full name.
+
+Now, let what happens when we call the command:
+
+```console
+$ vendor/bin/laminas example:hello
+Your name:
+ > Michal
+
+Hello Michal!
+```
+
+or, if you provide the param value with the command call:
+
+```console
+$ vendor/bin/laminas example:hello --name=Michal
+
+Hello Michal!
+```
+
+As you can see, there is no question asked when the parameter value is provided on command call.
+
+## Build-in types
+
+### InputParam::TYPE_BOOL
+
+```php
+$this->addParam(
+    'enable',
+    'Enable something',
+    InputParam::TYPE_BOOL,
+    false, // if the param is required, ignored for TYPE_BOOL
+    false // default value in interactive mode: true or false
+);
+```
+
+If the option is not set in command line call, in interactive mode use is going to be asked for confirmation:
+
+```console
+Enable something? [y/N]
+ > ...
+```
+
+Value of this parameter is normalized to `boolean` type.
+
+### InputParam::TYPE_INT
+
+```php
+$this->addParam(
+    'age',
+    'Your age',
+    InputParam::TYPE_INT,
+    true, // if the param is required (answer in interactive mode, if param value is not supplied)
+    null, // no default value
+    [
+        'min' => 0,
+        'max' => 120,
+    ]
+);
+```
+
+With InputParam::TYPE_INT we can define two options for validator: `min` and `max` value.
+Value of this parameter is normalized to `interger` type.
+
+### InputParam::TYPE_STRING
+
+```php
+$this->addParam(
+    'module',
+    'Module name',
+    InputParam::TYPE_STRING,
+    true, // if the param is required (answer in interactive mode, if param value is not supplied)
+    null, // no default value
+    [
+        'pattern' => '/^[A-Z][a-zA-Z0-9]{2,15}$/',
+    ]
+);
+```
+
+We can provide regular expression pattern to validate the value.
+In the above example we are expecting module name starting from an uppercase letter,
+then any letter or number, in total length of minimum 3 and maximum 16 characters.
+
+### InputParam::TYPE_PATH
+
+```php
+$this->addParam(
+    'dir',
+    'Directory with modules',
+    InputParam::TYPE_PATH,
+    true, // if the param is required (answer in interactive mode, if param value is not supplied)
+    'module',
+    [
+        'type' => 'dir', // 'dir', 'file' or null (for both)
```

Comment:
I would recommend referencing the constants you've defined (`PathValidator::TYPE_DIR`, `PathValidator::TYPE_FILE`), and maybe add another, `PathValidator::PATH_ANY`. Constants > magic strings.